custom_components/ha-glocaltokens/api.py

```diff
@@ -46,3 +52,53 @@ def get_google_devices_json(self):
     def get_android_id(self):
         """Generate random android_id"""
         return self._client._get_android_id()
+
+    def get_google_devices_information(self):
+
+        devices = self._client.get_google_devices_json()
+
+        for x in range(len(devices)):
+            #local_token = next(item[GLOCALTOKENS_TOKEN] for item in devices)
+            local_token = devices[0][GLOCALTOKENS_TOKEN] # Else x
+
+            url = 'https://IP OF DEVICE:8443/setup/assistant/alarms' # ONLY FOR TESTING PURPOSE - Until we find a way to retrieve ip of each device.
+            header = {'cast-local-authorization-token': local_token,
+                      'content-type': 'application/json'}
+
+            response = requests.get(url, headers=header, verify=False, timeout=TIMEOUT)
+
+            if response.status_code != 200:
+                _LOGGER.error("API returned {}".format(response.status_code))
+                #_LOGGER.error(local_token)
+                return devices
+
+            result = response.json()
+
+            if GLOCALTOKENS_TIMERS not in result and GLOCALTOKENS_ALARMS not in result:
+                _LOGGER.error("API returned unknown json structure")
+                return devices
+
+            devices[x][GLOCALTOKENS_TIMERS] = result[GLOCALTOKENS_TIMERS]
+            for y in range(len(result[GLOCALTOKENS_TIMERS])):
+
+                timestamp_ms = result[GLOCALTOKENS_TIMERS][y]['fire_time']
+                timestamp = timestamp_ms / 1000
+                humantime = datetime.fromtimestamp(timestamp).strftime('%H:%M:%S')
+                devices[x][GLOCALTOKENS_TIMERS][y]['date_time'] = humantime
+
+                duration_ms = result[GLOCALTOKENS_TIMERS][y]['original_duration']
+                duration = duration_ms / 1000
+                humanduration = datetime.utcfromtimestamp(duration).strftime('%H:%M:%S')
+                devices[x][GLOCALTOKENS_TIMERS][y]['duration'] = humanduration
+
+            devices[x][GLOCALTOKENS_ALARMS] = result[GLOCALTOKENS_ALARMS]
+            for z in range(len(result[GLOCALTOKENS_ALARMS])):
+
+                timestamp_ms = result[GLOCALTOKENS_ALARMS][z]['fire_time']
+                timestamp = timestamp_ms / 1000
+                humantime = datetime.utcfromtimestamp(timestamp).strftime('%Y-%m-%dT%H:%M:%S.%fZ%Z')
```

Comment:
The time format should also be constant so it would be easier to change if needed. I also wonder if HA don't have any pre-defined one, that would be even better, so we keep it consistent with them